### DISCUSSION TITLE
How to use ::before, ::after selector in Prismane?

### DISCUSSION THREAD

**Discussion by LuuKhoaHoc** [prismaneui/prismane#90] [Q&A] [answered]
I can not see this in Docs, someone help me? Question with love <3

**Comment by spleafy** [comment]
Hi, I am sorry to hear that you can't find it in the Docs. This is a mistake that we must and will fix soon by rewriting the docs. Sorry for the inconvenience!

To use pseudo classes, you can pass an array whose first element is the normal value and the second one is an object whose keys are the names of the pseudo-class.

For example, you can add hover like this:

m={16} -> m={[16, { hover: 32 }]}

You can also check out these few examples:

[https://www.prismane.io/docs/styling/style-props#style-prop-structure](https://www.prismane.io/docs/styling/style-props#style-prop-structure)

Sorry for the inconvenience and thanks for providing feedback, we will make sure to fix the Docs as soon as possible! <3

**Reply by LuuKhoaHoc** [reply]
I gonna try this right now. Thank you for reply <3

**Reply by spleafy** [reply]
Great, please, update me on the situation after you try it. I am here to assist with anything else too.

**Reply by LuuKhoaHoc** [reply]
Thank a lot

**Reply by LuuKhoaHoc** [reply]
I just try this again, it work with "hover" but i don't know how to do with "before" or "after".
I have a css hover animation like this: 
`a {
  position: relative;
}
a:hover {
  color: #39b54a;
}

a::before {
    content: '';
    position: absolute;
    width: 100%;
    height: 4px;
    border-radius: 4px;
    background-color: #39b54a;
    bottom: -5px;
    left: 0;
    transform-origin: right;
    transform: scaleX(0);
    transition: transform .2s ease-in-out;
  }

a:hover::before {
  transform-origin: left;
  transform: scaleX(1);`
  
  How to apply this with Prismane ?

**Reply by LuuKhoaHoc** [reply]
I have a Prismane code below to make a navbar for my project
`        
         <Box w={'100%'}>
          <Flex justify='around' align='center' h={'100%'}>
            <Box as={'a'} href='#' td={'none'} cl={['#fff', {hover: "primary"}]} pos={'relative'}>
              Tìm hiểu
            </Box>
            <Box as={'a'} href='#' td={'none'} cl={'#fff'}>
              Khuyến mãi{' '}
            </Box>
            <Box as={'a'} href='#' td={'none'} cl={'#fff'}>
              Đặt bàn{' '}
            </Box>
            <Box as={'a'} href='#' td={'none'} cl={'#fff'}>
              Đặt hàng{' '}
            </Box>
            <Box as={'a'} href='#' td={'none'} cl={'#fff'}>
              Thực đơn{' '}
            </Box>
            <Box as={'a'} href='#' td={'none'} cl={'#fff'}>
              Liên hệ
            </Box>
          </Flex>
        </Box>
`
    Thank you very much if you help me 🥲

**Reply by spleafy** [reply]
Pseudo-classes like ::before and ::after are usually added using the `sx` prop. I have reproduced the code that you provided and there seems to be an issue with the `sx` prop and pseudo-classes and how they are handled by Stitches. I will dig deeper and I will make sure to fix it immediately and update you when it is fixed.

For now, you can handle ::before and ::after with m={[16, { hover: 32, ":before": 24, ":after": 40 }]}

I am very sorry about this, I will test it right away!

**Reply by LuuKhoaHoc** [reply]
I love you bro <3. Keep it up!

**Reply by spleafy** [reply, marked as answer]
Hi! There is no issue in the `sx` prop, it works perfectly. To use ::before and ::after pseudo-classes, you can use the `sx` prop and prefix the pseudo-classes with `&`. So something like this:

`sx={{
  "&::before": {
    content: "''",
    display: "block",
    background: "red",
    color: "white",
  },
}}`

Please, try it out and update me when you have tried it, I would appreciate that! <3

**Reply by LuuKhoaHoc** [reply]
I will try right now 😍

**Reply by LuuKhoaHoc** [reply]
It works bro <3, thank a lot.

**Reply by spleafy** [reply]
Glad I helped! Thank you for your contribution too! I received a lot of meaningful and helpful feedback about the Docs from you, and I can assure you that I will fix them! Thank you <3
